### PR TITLE
SONARJAVA-1683 RSPEC-2148 Underscores on number check

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.INT_LITERAL, Tree.Kind.LONG_LITERAL);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    LiteralTree literalTree = (LiteralTree) tree;
+    if (!containsUnderscore(literalTree) && shouldUseUnderscore(literalTree.value())) {
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java]: Potential FP on `serialVersionUID`.

File: java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  }
+
+  private static boolean containsUnderscore(LiteralTree literalTree) {
+    return literalTree.value().indexOf('_') >= 0;
+  }
+
+  private static boolean shouldUseUnderscore(String literalValue) {
+    String value = LiteralUtils.trimLongSuffix(literalValue);
+    return Arrays.stream(Base.values())
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java]: Remove `minus` of negative values.

File: java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  private static boolean containsUnderscore(String literalValue) {
+    return literalValue.indexOf('_') >= 0;
+  }
+
+  private static boolean isSerialVersionUID(Tree tree) {
+    Tree parent = tree.parent();
+    while (parent != null && !parent.is(Tree.Kind.VARIABLE)) {
+      parent = parent.parent();
+    }
+    return parent != null && "serialVersionUID".equals(((VariableTree) parent).symbol().name());
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java]: Why going for semantic here ??

Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/UnderscoreOnNumberCheck.java]: Indeed I should just look at the identifier.

**Pull request merged**